org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/util/nodeselection/ActionShowingContextMenuListener.java

```diff
@@ -64,6 +69,10 @@ private void handleSelection(IMenuManager manager, NodeSelection selection) {
                 }
             }
         }
+
+        if (isEmpty) {
```

Comment:
The `isEmpty` variable is unnecessary. You can replace it with `this.actions.isEmpty()`.

Reply:
I think it is useful , cause "isEmpty" is for checking if no actions are visible for selected node.

Reply:
My bad, I thought that the `isEmpty = true` statement is outside of `if (action.isVisibleFor(selection)) {}`